pkg/provider/identity.go

```diff
@@ -35,26 +54,235 @@ type PeerIdentity struct {
 type FullIdentity struct {
 	PeerIdentity
 	PrivateKey crypto.PrivateKey
+}
 
-	todoCert *tls.Certificate // TODO(jt): get rid of this and only use the above
+type CAConfig struct {
+	CertPath   string `help:"path to the certificate chain for this identity" default:"$CONFDIR/identity.leaf.cert"`
+	KeyPath    string `help:"path to the private key for this identity" default:"$CONFDIR/identity.leaf.key"`
+	Difficulty uint16 `default:"24" help:"minimum difficulty for identity generation"`
+	Version    string `help:"semantic version of CA storage format"`
 }
 
 // IdentityConfig allows you to run a set of Responsibilities with the given
 // identity. You can also just load an Identity from disk.
 type IdentityConfig struct {
 	CertPath string `help:"path to the certificate chain for this identity" default:"$CONFDIR/identity.leaf.cert"`
 	KeyPath  string `help:"path to the private key for this identity" default:"$CONFDIR/identity.leaf.key"`
+	Version  string `help:"semantic version of identity storage format"`
 	Address  string `help:"address to listen on" default:":7777"`
 }
 
-// LoadIdentity loads a FullIdentity from the given configuration
-func (ic IdentityConfig) LoadIdentity() (*FullIdentity, error) {
-	pi, err := FullIdentityFromFiles(ic.CertPath, ic.KeyPath)
+// GenerateCA creates a new full identity with the given difficulty
+func GenerateCA(ctx context.Context, difficulty uint16, concurrency uint) *CertificateAuthority {
+	if ctx == nil {
+		ctx = context.Background()
+	}
```

Comment:
the go context style says you should never ever ever pass a context as nil, and thus you should never have to worry about checking if it's nil

Reply:
noted - i'll clean it up